### DISCUSSION TITLE
Addin Vue UI libraries to iles

### DISCUSSION THREAD

**Discussion by R0N1n-dev** [ElMassimo/iles#271] [General]
Have been having an issue trying to intergrate Vue Ui libraries especially the ones which contain JS (like Vuesax-alpha, Inkline, WaveuI) into iles. 
The CDN method with link and script tags presents an error saying "Tags with side-effects are ignored in client content templates" like below: 
![Screenshot from 2024-03-07 15-33-12](https://github.com/ElMassimo/iles/assets/83344872/2b924fa4-04ed-48b3-b472-8390ac06a4dc).

I tried using a plugin similar to how one would do while using Nuxt

vuesax.js
```js
import Vuesax from "vuesax-alpha";

export default Vuesax;
```

.I honestly don't know what to do about this. Help please.

**Comment by TechAkayy** [comment]
Check this thread, Massimo gave some good suggestions- https://github.com/ElMassimo/iles/discussions/267#discussioncomment-8212685.

**Reply by R0N1n-dev** [reply]
Have tried that
![Screenshot from 2024-03-07 15-59-01](https://github.com/ElMassimo/iles/assets/83344872/1c646995-0ac4-4af5-897a-aa71fd6ba3ec)
![Screenshot from 2024-03-07 16-00-02](https://github.com/ElMassimo/iles/assets/83344872/35f35842-1042-4653-a51a-3c3adbc60622)
I read it and have tries useScripttag and useHead with no success on the JS part, only the css is coming through. That's why the words are blue.
My issue is really the script tags. Not working at all

**Reply by R0N1n-dev** [reply]
The button  that is labelled "Button" should be looking like this
![Screenshot from 2024-03-07 16-05-12](https://github.com/ElMassimo/iles/assets/83344872/b8b43fd6-dd8b-421e-b3fe-00d02ffc1662)
 It was the same struggle with Astro.js, except in Astro, you can atleast mess with the head settings in the layout and do it directly

**Reply by TechAkayy** [reply]
Hi @R0N1n-dev, 

I have used useHead to load font urls. Please refer to this component in my sample repo - https://github.com/Pinegrow/pg-iles-vuetify/blob/main/src/components/HeadAndMeta.vue. This component is imported inside the default layout. 

If you have a reproduction, I can certainly have a look. That said, it does help (especially for new users) having a index.html shell like in this feature request - https://github.com/ElMassimo/iles/issues/136, @ElMassimo did say that it's planned in the other thread. 

I do have other iles & astro sample repos in the Pinegrow org (of the above example). Check it out. Here is an Astro example - https://github.com/Pinegrow/pg-astro-vuetify/blob/main/src/components/BaseHead.astro. I have say, I have had trouble working with Vue based UI libraries (specifically Vuetify) with Astro (https://github.com/withastro/astro/issues/8987), didn't get much support from the astro team on this topic.

**Reply by R0N1n-dev** [reply]
![Screenshot from 2024-03-08 09-07-35](https://github.com/ElMassimo/iles/assets/83344872/af65b56f-f709-4de3-9886-36346d82b8f5)
What am i doing wrong???????????????

**Reply by TechAkayy** [reply]
Can you try moving the useHead into default.vue.

**Reply by R0N1n-dev** [reply]
nothing. still the same

**Reply by R0N1n-dev** [reply]
It honestly should not be that hard to add a script to a project like this

**Reply by TechAkayy** [reply]
It should ideally work, must be something trivial. I can certainly help if you could push ur test repo to github and share

**Reply by TechAkayy** [reply]
Did you register inkline via `app.ts`? Source - https://www.inkline.io/docs/installation/cdn#starter-template 

In Iles, it's done in app.ts, here is an example - https://github.com/Pinegrow/pg-iles-vuetify/blob/main/src/app.ts. 

```
import { Inkline, components } from '@inkline/inkline';
// ... 


app.use(Inkline, { components });
```

**Reply by TechAkayy** [reply]
I'm going offline now as the library at my end is closing.. if you leave a msg with a github repo, I can check in the morning. All th best!

**Reply by TechAkayy** [reply]
Any luck? 🤔

**Comment by R0N1n-dev** [comment]
None

mdrewj - Chat @ Spike<https://spikenow.com/r/a/?ref=spike-organic-signature&_ts=2i34px> [2i34px]

On March 9, 2024 at 23:10 GMT, Akayy ***@***.******@***.***>> wrote:


Any luck? 🤔

—
Reply to this email directly, view it on GitHub<https://github.com/ElMassimo/iles/discussions/271#discussioncomment-8732448>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AT3332H7FI5DBJHOCDUSSITYXOJFXAVCNFSM6AAAAABEK6AKNKVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DOMZSGQ2DQ>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

[https://bolt.im/t/?etms6DgFJwu0tLk57XT-O-fUc71mBqfJPZhA1OQ1DLOS4FJ8ZPlsnsV1PC-I5lGQfvE0jOY8XIJ905inHdy5xw]

**Reply by R0N1n-dev** [reply]
No luck at all

**Reply by TechAkayy** [reply]
Is your repo shareable via github?

**Reply by R0N1n-dev** [reply]
Here is my repo.
https://stackblitz.com/github/R0N1n-dev/iles-pwa

**Reply by TechAkayy** [reply]
Thanks, I have replied here - https://github.com/ElMassimo/iles/issues/272#issuecomment-2014866913
